src/Course/Alternative.hs

```diff
@@ -0,0 +1,172 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE RebindableSyntax #-}
+
+module Course.Alternative where
+
+import Course.Applicative
+import Course.Core
+import Course.Functor
+import Course.List
+import Course.Optional
+import Course.Parser
+import qualified Prelude as P(fmap, return, (>>=))
+
+-- | All instances of the `Alternative` type-class must satisfy three laws.
+-- These laws are not checked by the compiler. These laws are given as:
+--
+-- * The law of left identity
+--   `∀x. empty <|> x = x`
+--
+-- * The law of right identity
+--   `∀x. x <|> empty = x`
+--
+-- * The law of associativity
+--   `∀u v w. u <|> (v <|> w) = (u <|> v) <|> w`
+--
+-- You may notice that these are the same laws as Monoid. An alternative
+-- can be considered a "monoid on applicative functors". The key difference
+-- between the two classes is that Alternative is higher-kinded, meaning that
+-- the type variable @k@ itself takes a type parameter.
+-- It is common for a type to have different instances for Monoid and
```

Comment:
A type `k` cannot be `Alternative` and `Monoid` because it won't have the correct kind. Maybe:
"The `Alternative` instance for `k` is often distinct from any `Monoid` instance for `k a`."?

To talk about the connection to `Applicative`, you need to mention the laws around `some` and `many`, which is challenging because those are exercises.